src/bootstrap/src/utils/helpers.rs

```diff
@@ -53,8 +54,27 @@ pub fn exe(name: &str, target: TargetSelection) -> String {
 }
 
 /// Returns `true` if the file name given looks like a dynamic library.
-pub fn is_dylib(name: &str) -> bool {
-    name.ends_with(".dylib") || name.ends_with(".so") || name.ends_with(".dll")
+pub fn is_dylib(path: &Path) -> bool {
+    path.extension().and_then(|ext| ext.to_str()).map_or(false, |ext| {
+        ext == "dylib" || ext == "so" || ext == "dll" || (ext == "a" && is_aix_shared_archive(path))
+    })
+}
+
+fn is_aix_shared_archive(path: &Path) -> bool {
+    // todo: reading the entire file as &[u8] into memory seems excessive
+    // look into either mmap it or use the &CacheReader
```

Comment:
Suggestion [AIX_SHARED_ARCHIVE_HEURISTICS 2/2]: can you convert this to a FIXME and open a `C-cleanup` issue then link to it `FIXME(#nnnnnn): ...`?